src/package-hoister.js

```diff
@@ -487,6 +487,14 @@ export default class PackageHoister {
   prepass(patterns: Array<string>) {
     patterns = this.resolver.dedupePatterns(patterns).sort();
 
+    const visited: {
```

Comment:
Would you mind changing this to use a `Map` instead? They're available starting from Node 4, so it should be fine for the compatibility, and I expect it to improve the perfs an extra bit :)